source/common/upstream/upstream_impl.cc

```diff
@@ -262,12 +262,34 @@ ResourceManager& ClusterInfoImpl::resourceManager(ResourcePriority priority) con
   return *resource_managers_.managers_[enumToInt(priority)];
 }
 
-void ClusterImplBase::blockHcUpdates(bool block) {
-  ASSERT(block_hc_updates_ == !block);
-  block_hc_updates_ = block;
+void ClusterImplBase::setInitialized() {
+  if (!initialized_ && health_checker_ && pending_initialize_health_checks_ == 0) {
+    pending_initialize_health_checks_ = hosts().size();
+    ASSERT(pending_initialize_health_checks_ > 0);
```

Comment:
Where in the config is having a health checker but no hosts illegal? Is it plausible someone would add a health checker for future additions to an  original_dst_cluster?

Reply:
Good catch. I think that's probably not required anymore. Will remove and add some more tests.